diff --git a/kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java b/kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java
index b0dd56b45d67..89d7878172cb 100644
--- a/kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java
+++ b/kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java
@@ -55,6 +55,7 @@
 import org.apache.iceberg.types.Types.StructType;
 import org.apache.iceberg.types.Types.TimeType;
 import org.apache.iceberg.types.Types.TimestampType;
+import org.apache.iceberg.types.Types.UUIDType;
 import org.apache.iceberg.util.Pair;
 import org.apache.iceberg.util.Tasks;
 import org.apache.kafka.connect.data.Date;
@@ -286,6 +287,10 @@ Type toIcebergType(Schema valueSchema) {
                   .collect(Collectors.toList());
           return StructType.of(structFields);
         case STRING:
+          if ("uuid".equals(valueSchema.name())) {
+            return UUIDType.get();
+          }
+          return StringType.get();
         default:
           return StringType.get();
       }
